src/main/java/io/gravitee/policy/groovy/GroovyPolicy.java

```diff
@@ -15,250 +15,177 @@
-                                        ex.getResult().getError(),
-                                        ex.getResult().getContentType()
-                                    )
-                                );
-                            } else {
-                                policyChain.streamFailWith(
-                                    io.gravitee.policy.api.PolicyResult.failure(
-                                        ex.getResult().getKey(),
-                                        ex.getResult().getCode(),
-                                        ex.getResult().getError()
-                                    )
-                                );
-                            }
-                        } catch (Throwable t) {
-                            logger.error("Unable to run Groovy script", t);
-                            throw new TransformationException("Unable to run Groovy script: " + t.getMessage(), t);
-                        }
-                        return null;
-                    }
-                )
-                .build();
+    @Override
+    public Completable onResponse(HttpExecutionContext ctx) {
+        if (configuration.isReadContent()) {
+            return onResponseContent(ctx);
         }
-
-        return null;
+        return runScript(ctx, GroovyBindings.bindHttp(ctx), configuration.getScript());
     }
 
-    @OnRequestContent
-    public ReadWriteStream onRequestContent(
-        Request request,
-        Response response,
-        ExecutionContext executionContext,
-        PolicyChain policyChain
-    ) {
-        String script = groovyPolicyConfiguration.getOnRequestContentScript();
+    private Completable onResponseContent(HttpExecutionContext ctx) {
+        return ctx
+            .response()
+            .onBody(bodyBuffer ->
+                bodyBuffer
+                    .defaultIfEmpty(Buffer.buffer())
+                    .flatMapMaybe(buffer ->
+                        onHttpContent(ctx, buffer, ctx.response().headers(), GroovyBindings.bindResponseContent(ctx, buffer))
+                    )
+            );
+    }
 
-        if (script != null && !script.trim().isEmpty()) {
-            return TransformableRequestStreamBuilder
-                .on(request)
-                .chain(policyChain)
-                .transform(
-                    buffer -> {
-                        try {
-                            final String content = executeStreamScript(
-                                new ContentAwareRequest(request, buffer.toString()),
-                                new ContentAwareResponse(response, null),
-                                executionContext,
-                                script
-                            );
+    private Maybe<Buffer> onHttpContent(HttpExecutionContext ctx, Buffer bodyBuffer, HttpHeaders headers, Binding binding) {
+        return scriptFlowable
+            .flatMapMaybe(script -> runContentAwareScript(ctx, binding, script))
```

Comment:
Ok, so I got the explanation regarding the `scriptFlowable` (to support v2 API in v4 emulation mode and have request/response and requestContent/responseContent scripts)

To mimic the behavior, you should ensure that the execution is sequential so probably use `concatMapMaybe` instead.